packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -240,6 +240,49 @@ String _generatePluralMethod(Message message, String translationForMessage) {
     .replaceAll('@(none)\n', '');
 }
 
+String _generateSelectMethod(Message message, String translationForMessage) {
+  if (message.placeholders.isEmpty) {
+    throw L10nException(
+      'Unable to find placeholders for the select message: ${message.resourceId}.\n'
+      'Check to see if the select message is in the proper ICU syntax format '
+      'and ensure that placeholders are properly specified.'
+    );
+  }
+
+  final List<String> cases = <String>[];
+
+  final RegExp selectRE = RegExp(r'\{([\w\s,]*),\s*select\s*,\s*([\w\d]+\s*\{.*\})+\s*\}');
```

Comment:
I would hoist this regex into a static so that it is only compiled once, and additional document a bit what it is intended to match to help future contributors.